elements/lisk-tree/src/merkle_tree.ts

```diff
@@ -154,200 +150,65 @@ export class MerkleTree {
 		return this.root;
 	}
 
-	public generatePath(_queryData: ReadonlyArray<Buffer>): Path {
+	public generateProof(queryData: ReadonlyArray<Buffer>): Proof | undefined {
 		if (this._width === 1) {
-			return [];
+			return undefined;
```

Comment:
should this return root as Proof or undefined is fine?

Reply:
I think it was left unspecified in the LIP, I would return a `Proof` with path=[], indexes=[], dataLength=1, because we are assuming the root is known